eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/sorted/mutable/TreeBag.java

```diff
@@ -444,17 +444,20 @@ public void value(T key, Counter value)
         return new InternalIterator();
     }
 
-    public void addOccurrences(T item, int occurrences)
+    public int addOccurrences(T item, int occurrences)
     {
         if (occurrences < 0)
         {
             throw new IllegalArgumentException("Cannot add a negative number of occurrences");
         }
         if (occurrences > 0)
         {
-            this.items.getIfAbsentPut(item, NEW_COUNTER_BLOCK).add(occurrences);
+            Counter updatedOccurences = this.items.getIfAbsentPut(item, NEW_COUNTER_BLOCK);
```

Comment:
Since updatedOccurrences is mutable and gets updated on the next line, I'd name it `counter` or `occurrences` instead.